pandas/tests/test_reshape.py

```diff
@@ -716,6 +716,81 @@ def test_stubs(self):
+        df["id"] = df.index
+        exp_data = {"X": x.tolist() + x.tolist(),
+                    "A": ['a', 'b', 'c', 'd', 'e', 'f'],
+                    "B": [2.5, 1.2, 0.7, 3.2, 1.3, 0.1],
+                    "year": ['1970', '1970', '1970', '1980', '1980', '1980'],
+                    "id": [0, 1, 2, 0, 1, 2]}
+        exp_frame = DataFrame(exp_data)
+        exp_frame = exp_frame.set_index(['id', 'year'])[["X", "A", "B"]]
+        long_frame = wide_to_long(df, ["A.", "B."], i="id", j="year", sep=".")
+        tm.assert_frame_equal(long_frame, exp_frame)
+
+    def test_escapable_characters(self):
+        np.random.seed(123)
+        x = np.random.randn(3)
+        df = pd.DataFrame({"A (quarterly) 1970": {0: "a",
+                                                  1: "b",
+                                                  2: "c"},
+                           "A (quarterly) 1980": {0: "d",
+                                                  1: "e",
+                                                  2: "f"},
+                           "B (quarterly) 1970": {0: 2.5,
+                                                  1: 1.2,
+                                                  2: .7},
+                           "B (quarterly) 1980": {0: 3.2,
+                                                  1: 1.3,
+                                                  2: .1},
+                           "X": dict(zip(
+                               range(3), x))})
+        df["id"] = df.index
+        exp_data = {"X": x.tolist() + x.tolist(),
+                    "A (quarterly)": ['a', 'b', 'c', 'd', 'e', 'f'],
+                    "B (quarterly)": [2.5, 1.2, 0.7, 3.2, 1.3, 0.1],
+                    "year": ['1970', '1970', '1970', '1980', '1980', '1980'],
+                    "id": [0, 1, 2, 0, 1, 2]}
+        exp_frame = DataFrame(exp_data)
+        exp_frame = exp_frame.set_index(
+            ['id', 'year'])[["X", "A (quarterly)", "B (quarterly)"]]
+        long_frame = wide_to_long(df, ["A (quarterly) ", "B (quarterly) "],
+                                  i="id", j="year", sep=" ")
+        tm.assert_frame_equal(long_frame, exp_frame)
+
+    def test_unbalanced(self):
+        # test that we can have a varying amount of time variables
+        df = pd.DataFrame({'A2010': [1.0, 2.0],
+                           'A2011': [3.0, 4.0],
+                           'B2010': [5.0, 6.0],
+                           'X': ['X1', 'X2']})
+        df['id'] = df.index
+        exp_data = {'X': ['X1', 'X1', 'X2', 'X2'],
+                    'A': [1.0, 3.0, 2.0, 4.0],
+                    'B': [5.0, np.nan, 6.0, np.nan],
+                    'id': [0, 0, 1, 1],
+                    'year': ['2010', '2011', '2010', '2011']}
+        exp_frame = pd.DataFrame(exp_data)
+        exp_frame = exp_frame.set_index(['id', 'year'])[["X", "A", "B"]]
+        long_frame = wide_to_long(df, ['A', 'B'], 'id', 'year')
+        tm.assert_frame_equal(long_frame, exp_frame)
+
```

Comment:
What input where you thinking about?

if a nonsense separator is passed nothing is stripped:

```python
In [15]: df = pd.DataFrame({'A.2010': np.random.rand(3),
    ...:                    'A.2011': np.random.rand(3),
    ...:                    'B.2010': np.random.rand(3),
    ...:                    'B.2011': np.random.rand(3),
    ...:                    'X' : np.random.randint(3, size=3)})
    ...: df['id'] = df.index
    ...: pd.wide_to_long(df, ['A.', 'B.'], i='id', j='year', sep="nope")
    ...:
Out[15]:
         X        A.        B.
id year
0  2010  2  0.330193  0.728615
1  2010  0  0.710791  0.601923
2  2010  1  0.066218  0.618455
0  2011  2  0.597949  0.324131
1  2011  0  0.024911  0.968051
2  2011  1  0.310596  0.866798
```

```python
In [16]: pd.wide_to_long(df, ['A.', 'B.'], i='id', j='year', sep=",,")
Out[16]:
         X        A.        B.
id year
0  2010  2  0.330193  0.728615
1  2010  0  0.710791  0.601923
2  2010  1  0.066218  0.618455
0  2011  2  0.597949  0.324131
1  2011  0  0.024911  0.968051
2  2011  1  0.310596  0.866798
```